include/aws/s3/s3_client.h

```diff
@@ -109,6 +109,25 @@ enum aws_s3_meta_request_compute_content_md5 {
     AWS_MR_CONTENT_MD5_ENABLED,
 };
 
+enum aws_s3_checksum_algorithm {
+    AWS_CRC32C,
```

Comment:
I would put a prefix on these: AWS_SOMETHING_ALGORITHM.  I usually just take the first letters of the word (SCA here) but anything reasonable should be fine.